pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/renderer/TextValueProvider.java

```diff
@@ -1,101 +1,224 @@
+        return Collections.singletonList(formatRawValues(rawValues, filterConfig, formMap));
     }
 
-    protected Object getRawAnswerValue(LinkedHashMap<String, Object> fq, String columnName) {
-        Object rawAnswer = fq.getOrDefault(ESObjectConstants.ANSWER, fq.get(columnName));
 
-        Collection<?> answer = mapToCollection(rawAnswer);
-        Object optionDetails = fq.get(ESObjectConstants.OPTIONDETAILS);
-        if (optionDetails != null && !((List<?>) optionDetails).isEmpty()) {
-            removeOptionsFromAnswer(answer, ((List<Map<String, String>>) optionDetails));
-            return Stream.of(answer, getOptionDetails((List<?>) optionDetails)).flatMap(Collection::stream)
-                    .collect(Collectors.toList());
+
+    protected List<String> formatRawValues(List<?> rawValues, FilterExportConfig filterConfig, Map<String, Object> formMap) {
+        return rawValues.stream().map(val -> {
+            if (val == null) {
+                return StringUtils.EMPTY;
+            } else if (val instanceof List) {
+                return (String) ((List) val).stream().map(item -> item != null ? item.toString() : StringUtils.EMPTY)
+                        .collect(Collectors.joining(", "));
+            }
+            return val.toString();
+        }).collect(Collectors.toList());
+    }
+
+    /** handles reading values from things other than questions. including basic properties and array properties
+     * it falls back to reading the dynamic fields if an object is specified that doesn't exist on the
+     * is found
+     */
+    protected Object getValueFromMap(Map<String, Object> moduleMap, FilterExportConfig filterConfig) {
+        Map<String, Object> targetMap = null;
+        String objectName = filterConfig.getColumn().getObject();
+        String fieldName = filterConfig.getColumn().getName();
+        if (objectName != null) {
+            Object formObject = moduleMap.get(objectName);
+            if (formObject instanceof List) {
+                // this is a list, like "testResult" off of kitRequestShipping
+                // transform an array of maps into a map of arrays
+                Map<String, Object> answerMap = new HashMap();
+                final String finalFieldName = fieldName;
+                List<Object> answersArray = ((List<?>) formObject).stream().map(arrValue -> {
+                    if (arrValue instanceof Map) {
+                        return ((Map<String, Object>) arrValue).get(finalFieldName);
+                    }
+                    return StringUtils.EMPTY;
+                }).collect(Collectors.toList());
+                answerMap.put(fieldName, answersArray);
+                targetMap = answerMap;
+
+            } else if (formObject instanceof Map) {
+                targetMap = (Map<String, Object>) formObject;
+            }
         }
+        if ((targetMap == null && objectName != null) || ESObjectConstants.ADDITIONAL_VALUE.equals(filterConfig.getType())) {
```

Comment:
having method with name `isDynamicFields` is more straightforward than having comment in my opinion